shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/decorator/HikariJDBCParameterDecorator.java

```diff
@@ -27,23 +28,33 @@
     
     @Override
     public HikariDataSource decorate(final HikariDataSource dataSource) {
-        dataSource.getDataSourceProperties().setProperty("useServerPrepStmts", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("cachePrepStmts", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("prepStmtCacheSize", "200000");
-        dataSource.getDataSourceProperties().setProperty("prepStmtCacheSqlLimit", "2048");
-        dataSource.getDataSourceProperties().setProperty("useLocalSessionState", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("rewriteBatchedStatements", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("cacheResultSetMetadata", Boolean.FALSE.toString());
-        dataSource.getDataSourceProperties().setProperty("cacheServerConfiguration", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("elideSetAutoCommits", Boolean.TRUE.toString());
-        dataSource.getDataSourceProperties().setProperty("maintainTimeStats", Boolean.FALSE.toString());
-        dataSource.getDataSourceProperties().setProperty("netTimeoutForStreamingResults", "0");
-        dataSource.getDataSourceProperties().setProperty("tinyInt1isBit", Boolean.FALSE.toString());
+        addJDBCProperty(dataSource, "useServerPrepStmts", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "useServerPrepStmts", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "cachePrepStmts", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "prepStmtCacheSize", "200000");
+        addJDBCProperty(dataSource, "prepStmtCacheSqlLimit", "2048");
+        addJDBCProperty(dataSource, "useLocalSessionState", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "rewriteBatchedStatements", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "cacheResultSetMetadata", Boolean.FALSE.toString());
+        addJDBCProperty(dataSource, "cacheServerConfiguration", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "elideSetAutoCommits", Boolean.TRUE.toString());
+        addJDBCProperty(dataSource, "maintainTimeStats", Boolean.FALSE.toString());
+        addJDBCProperty(dataSource, "netTimeoutForStreamingResults", "0");
+        addJDBCProperty(dataSource, "tinyInt1isBit", Boolean.FALSE.toString());
+        addJDBCProperty(dataSource, "useSSL", Boolean.FALSE.toString());
+        addJDBCProperty(dataSource, "serverTimezone", "UTC");
```

Comment:
I read some code of MySQL JDBC Driver 5.1.47. There is a doc for `com.mysql.jdbc.NonRegisteringDriver#connect`wrote
>  The java.util.Properties argument can be used to pass arbitrary string tag/value pairs as connection arguments. These properties take precedence over any properties sent in the URL.

My understanding is that the parameters such as `useSSL=true` we write in the URL will not work.

Reply:
Thanks @TeslaCN .
I have read `parseURL` method of class `NonRegisteringDriver` too.  So I add a 'url check' in `addJDBCProperty()`, please have a see.

Reply:
I think using `indexOf` is too risk. Is there a better way?

Reply:
How about using `java.net.URI` to get the query part?